### ISSUE TITLE
Detected include path of vcpkg is wrong

### ISSUE THREAD

**Issue by TTsdzb** [twistedfall/opencv-rust#640]
Before reporting an issue please first check the
[troubleshooting guide](https://github.com/twistedfall/opencv-rust/blob/master/README.md#troubleshooting). If
the issue you're encountering is not solved thereby please state the following in your bugreport:
1. Operating system: Windows 11 23H2
2. The way you installed OpenCV: [vcpkg](https://github.com/twistedfall/opencv-rust/blob/master/INSTALL.md#windows-package)
3. OpenCV version: opencv4:x64-windows-static-md 4.10.0#1
4. rustc version (`rustc --version`): rustc 1.83.0 (90b35a623 2024-11-26)
5. Attach the full output of the following command from your project directory: (too long for issue) https://pastebin.com/ecKUPtbr
   ```shell script
   RUST_BACKTRACE=full cargo build -vv 
   ```

**Comment by TTsdzb** [comment]
The `include_path` should be `"C:\\Tools\\vcpkg\\installed\\x64-windows-static-md\\include\\opencv4"`, however the detected path is `"C:\\Tools\\vcpkg\\installed\\x64-windows-static-md\\include"`.

**Comment by TTsdzb** [comment]
Copying `include\\opencv4\\opencv2` to `include\\opencv2` will work, however this does not look like a good enough solution.

Besides, I ran into another linkage problem:

```
= note: libopencv-36ef7a0166df0d9b.rlib(65f8cfa25828e555-core.o) : error LNK2019: unresolved external symbol "public: bool __cdecl cv::ocl::Kernel::run_(int,unsigned __int64 * const,unsigned __int64 * const,bool,class cv::ocl::Queue const &)" (?run_@Kernel@ocl@cv@@QEAA_NHQEA_K0_NAEBVQueue@23@@Z)，referenced in function cv_ocl_Kernel_run__int_size_tX_size_tX_bool_const_QueueR
          libopencv-36ef7a0166df0d9b.rlib(65f8cfa25828e555-core.o) : error LNK2019: unresolved external symbol "public: int __cdecl cv::ocl::PlatformInfo::versionMajor(void)const " (?versionMajor@PlatformInfo@ocl@cv@@QEBAHXZ)，referenced in function cv_ocl_PlatformInfo_versionMajor_const
          libopencv-36ef7a0166df0d9b.rlib(65f8cfa25828e555-core.o) : error LNK2019: unresolved external symbol "public: int __cdecl cv::ocl::PlatformInfo::versionMinor(void)const " (?versionMinor@PlatformInfo@ocl@cv@@QEBAHXZ)，referenced in function cv_ocl_PlatformInfo_versionMinor_const
          libopencv-36ef7a0166df0d9b.rlib(65f8cfa25828e555-core.o) : error LNK2019: unresolved external symbol "char const * __cdecl cv::ocl::convertTypeStr(int,int,int,char *,unsigned __int64)" (?convertTypeStr@ocl@cv@@YAPEBDHHHPEAD_K@Z)，referenced in function cv_ocl_convertTypeStr_int_int_int_charX_size_t
          D:\liusq_workspace\auto-helper-rs\target\debug\deps\auto_helper_rs-d9772966fe176c07.exe : fatal error LNK1120: 4 unresolved externals


error: could not compile `auto-helper-rs` (lib test) due to 1 previous error
```

This seems like something within OpenCV itself and resolution in [TROUBLESHOOTING.md](https://github.com/twistedfall/opencv-rust/blob/master/TROUBLESHOOTING.md) does not work, either.

**Comment by TTsdzb** [comment]
For the linkage problem, the vcpkg must be installed with `opencl` feature:

```diff
- vcpkg install llvm opencv4[contrib,nonfree]
+ vcpkg install llvm opencv4[contrib,nonfree,opencl]
```

Please update the [INSTALL.md](https://github.com/twistedfall/opencv-rust/blob/master/INSTALL.md).

**Comment by twistedfall** [comment]
Can you please tell me the git rev hash of the vcpkg repo you're using at the moment you install `vcpkg`?

It looks like there is a problem with that install. I could add some hacks to try to detect `opencv4` inside the `include_path` detected by the `vcpkg`, but I wonder if it's needed. Your install will seemingly not work if you try to build the C++ app linking to OpenCV for example.

In the CI I'm testing with `2024.09.30` tag (https://github.com/microsoft/vcpkg/tree/2024.09.30) at the moment so I wonder if your version is older or newer.

**Comment by TTsdzb** [comment]
> Can you please tell me the git rev hash of the vcpkg repo you're using at the moment you install `vcpkg`?

Latest in the master branch, that is [cd124b8](https://github.com/microsoft/vcpkg/commit/cd124b84feb0c02a24a2d90981e8358fdee0e077).

**Comment by twistedfall** [comment]
Yep, it's definitely broken in the `vcpkg` on Windows on master and the `2024.11.16` tag. For now you can try rolling back to `2024.09.30` tag of `vcpkg`. I will investigate.

**Comment by 1mZ1kk4d0** [comment]
twistedfall the link error is really happening, it is not something directly related to the installation, in my case I need opencl too, I recommend updating the readme, but in my case I added additional libs such as world + ffmpeg + cuda

**Comment by twistedfall** [comment]
This should now be fixed in v0.93.7, there is now a workaround that patches up the incorrect include path from vcpkg
